src/lib/y2partitioner/widgets/delete_disk_partition_button.rb

```diff
@@ -48,7 +52,42 @@ def handle
 
     private
 
+      def confirm(device)
+        names = children_names(device)
+
+        if names.empty?
+          # TODO: check for children that will die and if there are, use confirm_recursive_delete
+          Yast::Popup.YesNo(
+            # TRANSLATORS %s is device to be deleted
+            format(_("Really delete %s?"), device.name)
+          )
+        else
+          confirm_recursive_delete(
+            names,
+            _("Confirm Deleting of All Partitions"),
+            format(_("The %{type} \"%{name}\" contains at least one another device.\n" \
+              "If you proceed, the following devices will be deleted:"),
+              name: device.name,
+              type: device.is?(:disk) ? _("disk") : _("partition")),
```

Comment:
in future I will ahve to extend it by lvm or raid, so it is just first step...and for sure dasd will need also improvement